File: 0768-partition-labels/0768-partition-labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        int[] last=new int[26];
        List<Integer> ans=new ArrayList<>();

        for(int i=0; i<s.length(); i++){
            last[s.charAt(i)-'a']=i;
        }
        int end=0, start=0;

        for(int i=0; i<s.length(); i++){
            end=Math.max(end,last[s.charAt(i)-'a']);

            if(i==end){
                ans.add(end-start+1);
                start=i+1;
                end=0;
            }
        }
        return ans;
    }
}

File: 0768-partition-labels/0768-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        n,l=len(s),0
        ans=[]*n

        while(l<n):
            r=s.rfind(s[l]) #Last Character
            i=l

            while(i<r):
                r=max(r,s.rfind(s[i]))
                i+=1
            ans.append(r-l+1)
            l=r+1
            
        return ans


        